backend/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -630,15 +631,21 @@ def setup_repo(self, repo, uln_repo=False):
         zypper_cmd = "zypper"
         if not self.interactive:
             zypper_cmd = "{} -n".format(zypper_cmd)
-        ret_error = os.system("{} --root {} --reposd-dir {} --cache-dir {} --raw-cache-dir {} --solv-cache-dir {} ref".format(
+        zypper_cmd = "{} --root {} --reposd-dir {} --cache-dir {} --raw-cache-dir {} --solv-cache-dir {} ref".format(
             zypper_cmd,
             REPOSYNC_ZYPPER_ROOT,
             os.path.join(repo.root, "etc/zypp/repos.d/"),
             REPOSYNC_ZYPPER_RPMDB_PATH,
             os.path.join(repo.root, "var/cache/zypp/raw/"),
             os.path.join(repo.root, "var/cache/zypp/solv/")
-        ))
+        )
+        process = subprocess.Popen(zypper_cmd.split(' '), stdout=subprocess.DEVNULL, stderr=subprocess.PIPE)
```

Comment:
Hmmm, I think this is going to cause some problems.

IIUC this will hide the stdout from "zypper ref" execution. Currently we see Zypper output refreshing the repo metadata when running `spacewalk-repo-sync`.

And even more problematic is that with this way I think we won't be able to accept the GPG keys in case Zypper is asking for it.

Maybe what you want is something like:

```python
   process = subprocess.run(zypper_cmd.split(' '), stderr=subprocess.PIPE)
```

This way, the output from Zypper will be shown and you can also capture retcode and stderr at `process.returncode` and `process.stderr`.

Does it make more sense?

Reply:
changed, please check